### DISCUSSION TITLE
Does `--rolling-restart` restart one by one replicas ?

### DISCUSSION THREAD

**Discussion by MaxR522** [containrrr/watchtower#2040] [Q&A]
I am running an application with 2 replicas, and I want to ensure zero downtime when upgrading the Docker image. Specifically, I want Watchtower to update the replicas one by one, so that at least one replica remains running at all times.

The documentation states:
"Restart one image at a time instead of stopping and starting all at once."

However, since my replicas are based on the same image, it's unclear to me whether the `--rolling-restart` option restarts each replica individually or if it still results in downtime for the entire service. Could you clarify if this option ensures one-by-one rolling updates for replicas?

```
services:
  webapp:
    image: myapp:latest
    deploy:
      replicas: 2 
```

**Comment by kinduff** [comment]
Can confirm it does. It will go one by one, can be seen here https://github.com/containrrr/watchtower/blob/76f9cea516593fabb8ca91ff13de55caa6aa0a8b/internal/actions/update.go#L97-L121
